charity-data/getcharities.py

```diff
@@ -0,0 +1,80 @@
+from urllib import parse
+import requests
+import json
+
+
+def get_charity_link(base_charity_url):
+    # Append / if not already present
+    if base_charity_url[len(base_charity_url) - 1] != '/':
+        base_charity_url += '/'
+    # If /donate is valid
+    if requests.get(base_charity_url + 'donate').status_code < 400:
```

Comment:
should we check for exact status code, like 200, 302 etc instead of generic condition.

Reply:
We could check for specific conditions but it'd be difficult to know which status codes to look for. Any code under 400 indicates a valid link (Informational responses (100–199), Successful responses (200–299), Redirects (300–399)) according to [https://developer.mozilla.org/en-US/docs/Web/HTTP/Status](url).

Reply:
You should check for Status code 200 OK. the success responses are in the ange 200-299 and 200 is the only relevant one here.

Reply:
Done.